### PR TITLE
fix: [GTM-541]: Send signup failure event to marketo (AND style fixes)

### PR DESCRIPTION
**Pull request by thomas-grasman**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31898)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/ServiceVariableCreator.java [hunk 1]
```diff
@@ -91,6 +91,7 @@ private void addVariablesForServiceSpec(YamlField serviceDefNode, Map<String, Ya
           }
           break;
         case ServiceSpecType.SSH:
+        case ServiceSpecType.WINRM:
```
Comment (pranil-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/ServiceVariableCreator.java:94]: @thomas-grasman - is this related to the change? It doesnt seem like a style fix either.

Comment (thomas-grasman) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/ServiceVariableCreator.java:94]: This is just from changes I pulled in after rebasing with develop. 
All my changes are in [920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java]
The only relevant changes are between lines 538 and 544. The rest are just style fixes marked critical by SonarLint.


[920-ng-signup/src/test/java/io/harness/signup/services/SignupServiceImplTest.java]
I also updated the tests because now we expect there to be 2 telemetry events instead of 1.

Comment (thomas-grasman) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/ServiceVariableCreator.java]: This is now fixed.

**Review by desagnik — DISMISSED** [review]
Looks fine to me.

**Review by nanaxuHarness — DISMISSED** [review]
> Reviewed 1 of 74 files at r1, 73 of 73 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31898)* status: 1. unresolved discussions.

**Review by pranil-harness — DISMISSED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31898)* status: :shipit: complete! ✓

**Comment by allenthom** [comment]
*[920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java, line 544 at r3](https://reviewable.io/reviews/harness/harness-core/31898#-N-dguApE7bmPt0FNbyB:-N-dguAq2VqA5OFkRS0x:b8ok4wz) ([raw file](https://github.com/harness/harness-core/blob/9421cee63237b8af81bb7d804eb70923ec57b139/920-ng-signup/src/main/java/io/harness/signup/services/impl/SignupServiceImpl.java#L544)):*
> ```Java
>     telemetryReporter.sendTrackEvent(FAILED_EVENT_NAME, email, accountIdentifier, properties,
>         ImmutableMap.<Destination, Boolean>builder().put(Destination.SALESFORCE, true).build(), Category.SIGN_UP);
>     telemetryReporter.sendTrackEvent(FAILED_EVENT_NAME, email, accountIdentifier, properties,
> ```

Instead of sending 2 different track events, just add an extra destination to one of them.
Ex. ImmutableMap.<Destination, Boolean>builder().put(Destination.SALESFORCE, true).put(Destination.MARKETO, true).build()

**Review by allenthom — COMMENTED** [review]
> Reviewed 1 of 74 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31898)* status: 0 of 1 approvals obtained.

**Review by pranil-harness — APPROVED** [review]
> Reviewed 1 of 74 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31898)* status: :shipit: complete! ✓
